core/src/main/scala/io/qbeast/spark/index/SparkRevisionFactory.scala

```diff
@@ -283,15 +300,26 @@ trait SparkRevisionChangesUtils extends StagingUtils with Logging {
    */
   private[index] def computeTransformationsFromColumnStats(
       transformers: IISeq[Transformer],
-      options: QbeastOptions): IISeq[Option[Transformation]] = {
-    val (columnStats, availableColumnStats) = parseColumnStats(options)
+      options: QbeastOptions,
+      dataSchema: StructType): IISeq[Option[Transformation]] = {
+    // 1. Get the columnStats from the options
+    val columnStatsString = options.columnStats.getOrElse("")
+    // 2. Build the QbeastColumnStats
+    val qbeastColumnStats =
+      QbeastColumnStatsBuilder.build(columnStatsString, transformers, dataSchema)
+    // 3. Compute transformations from the columnStats
+    val columnStatsRow = qbeastColumnStats.columnStatsRow
     transformers.map { t =>
-      if (t.stats.statsNames.forall(availableColumnStats.contains)) {
+      try {
```

Comment:
This logic here can be placed in a method in `QbeastColumnStats.`
In this way, the logic will be more clear:
1. We create an object of QbeastColumnStats
2. Then we use this object to create `Transformations.` If the columns are absent in the stats, we log a warning and return `None.`

Reply:
With that, we can also reduce the number of exceptions thrown here.